src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -34,157 +37,160 @@ type Props = {
+            <div className="bcs-task-assignments">
+                {assigneeCount > 0 &&
+                    assignees.entries
+                        .map(fillUserPlaceholder)
+                        .slice(0, MAX_AVATARS)
+                        .map(({ id: assignmentId, target, status: assigneeStatus }) => {
+                            return (
+                                <AssigneeStatus
+                                    key={assignmentId}
+                                    status={assigneeStatus}
+                                    user={target}
+                                    getAvatarUrl={getAvatarUrl}
+                                    data-testid="task-assignment-status"
+                                />
+                            );
+                        })}
+                {hiddenAssigneeCount > 0 ? (
+                    <span className="bcs-task-assignment-avatar bcs-task-assignment-count">+{hiddenAssigneeCount}</span>
+                ) : null}
+            </div>
+        </div>
+    );
+});
+
+const getMessageForTask = (isCurrentUser: boolean, taskType: TaskType) => {
+    if (isCurrentUser) {
         if (taskType === TASK_TYPE_APPROVAL) {
-            return messages.tasksFeedHeadlineApproval;
+            return messages.tasksFeedHeadlineApprovalCurrentUser;
         }
-        return messages.tasksFeedHeadlineGeneral;
+        return messages.tasksFeedHeadlineGeneralCurrentUser;
     }
 
-    render(): React.Node {
-        const {
-            assigned_to,
-            created_at,
-            created_by,
-            currentUser,
-            due_at,
-            error,
-            getAvatarUrl,
-            getMentionWithQuery,
-            getUserProfileUrl,
-            id,
-            isPending,
-            mentionSelectorContacts,
-            name,
-            onAssignmentUpdate = noop,
-            onDelete,
-            onEdit,
-            permissions,
-            status,
-            task_type,
-            translatedTaggedMessage,
-            translations,
-        } = this.props;
-
-        const taskPermissions = {
-            ...permissions,
-            // can_edit: permissions.can_update,
-            can_edit: false,
-        };
+    if (taskType === TASK_TYPE_APPROVAL) {
+        return messages.tasksFeedHeadlineApproval;
+    }
+    return messages.tasksFeedHeadlineGeneral;
+};
 
-        // // find assignment for current user if permitted
+const Task = ({
+    assigned_to,
+    created_at,
+    created_by,
+    currentUser,
+    due_at,
+    error,
+    getAvatarUrl,
+    getMentionWithQuery,
+    getUserProfileUrl,
+    id,
+    isPending,
+    mentionSelectorContacts,
+    name,
+    onAssignmentUpdate = noop,
+    onDelete,
+    onEdit,
+    permissions,
+    status,
+    task_type,
+    translatedTaggedMessage,
+    translations,
+}: Props) => {
+    const taskPermissions = {
+        ...permissions,
+        // can_edit: permissions.can_update,
```

Comment:
Can remove? Or should `can_edit` be modified?

Reply:
This will be implemented in a followup